Core/Core/Courses/CourseListView.swift

```diff
@@ -0,0 +1,178 @@
+        }
+    }
+
+    var empty: some View {
+        EmptyViewRepresentable(
+            title: NSLocalizedString("No Courses", comment: ""),
+            body: NSLocalizedString("It looks like there aren’t any courses associated with this account. Visit the web to create a course today.", comment: ""),
+            imageName: "PandaTeacher"
+        )
+    }
+
+    var courseList: some View {
+        // TODO: better searching
+        let filteredCourses = allCourses.filter { filter.isEmpty || $0.name?.lowercased().contains(filter.lowercased()) == true }
+        let currentEnrollments = filteredCourses.filter { !$0.isPastEnrollment && !$0.isFutureEnrollment }
+        let pastEnrollments = filteredCourses.filter { $0.isPastEnrollment }
+        let futureEnrollments = filteredCourses.filter { $0.isFutureEnrollment }
+        return VStack(spacing: 0) {
+            SearchBarView(text: $filter)
+            if filteredCourses.isEmpty {
+                Text("No matching courses").frame(maxHeight: .infinity)
+            } else {
+                Form {
+                    enrollmentSection(Text("Current Enrollments"), courses: currentEnrollments)
+                    enrollmentSection(Text("Past Enrollments"), courses: pastEnrollments)
+                    enrollmentSection(Text("Future Enrollments"), courses: futureEnrollments)
+                }
+            }
+        }
+    }
+
+    @ViewBuilder
+    func enrollmentSection<Header: View>(_ header: Header, courses: [Course]) -> some View {
+        if !courses.isEmpty {
+            Section(header: header) {
+                ForEach(courses, id: \.self) { course in
+                    Cell(course: course) {
+                        guard let controller = self.controller() else { return }
+                        self.env.router.route(to: "/courses/\(course.id)", from: controller)
+                    }
+                    .listRowInsets(EdgeInsets(top: 16, leading: 18, bottom: 16, trailing: 18))
+                }
+            }
+        }
+    }
+
+    struct Cell: View {
+        @Environment(\.appEnvironment) var env: AppEnvironment
+        @ObservedObject var course: Course
+        @State var pending = false
+        let didSelect: () -> Void
+
+        var body: some View {
+            ZStack {
+                Button(action: didSelect) { SwiftUI.EmptyView() }
+                HStack {
+                    favoriteButton
+                    label
+                    Spacer()
+                    publishedIcon
+                }
+            }
+        }
+
+        var favoriteButton: some View {
+            Button(action: toggleFavorite) {
+                if pending {
+                    Image.icon(.star, .solid).foregroundColor(.named(.ash))
+                } else if course.isFavorite {
+                    Image.icon(.star, .solid).foregroundColor(.named(.electric))
+                } else {
+                    Image.icon(.star, .line).foregroundColor(.named(.ash))
```

Comment:
I'd prefer these colors use their functional names. `.textDark`, `.textInfo`. Text colors are often used by icons. The `text` prefix would be more appropriate as `foreground`.